### PR TITLE
Support ingestion from multiple directories

### PR DESCRIPTION
**Pull request by apoorva918**
This PR adds support for user to ingest files from multiple folders. 
The directory paths can be separated by commas and passed to FILE_SPEC parameter. The directory paths will be iterated over, and each directory will be recursively scanned (in case of subdirectories) for files with the provided FILE_EXTENSION to obtain a list of files. The list of files will then be ingested.

Code for getting list of files from directories has been moved to io.pravega.sensor.collector.util.FileUtils.java
FileNameWithOffset.java and PravegaWriterEvent.java have also been refactored into io.pravega.sensor.collector.util package.

### CODE REVIEW

File: pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/parquet/ParquetFileProcessor.java [hunk 1]
```diff
@@ -137,12 +137,23 @@ protected List<FileNameWithOffset> getDirectoryListing() throws IOException {
      * @return list of file name and file size in bytes
      */
     static protected List<FileNameWithOffset> getDirectoryListing(String fileSpec, String fileExtension) throws IOException {
-        final Path pathSpec = Paths.get(fileSpec);
+        String[] directories= fileSpec.split(",");
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/parquet/ParquetFileProcessor.java:140]: Please put "," as a constant at the top

Comment (sachin-j-joshi) [reply pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/parquet/ParquetFileProcessor.java:140]: The file/directory scanning code in both ParqueFileProcessor and RawFileProcessor seems to be identical. 

I suggest either of these - 
1. Create a class FileUtils under utils folder and move the methods there.
2. Create a base class like BaseFileProcessor and move this functionality there

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
Please address the comments and also add unit test to test the functionality.

File: pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/parquet/ParquetFileProcessor.java (hunk 1 above)
Comment (apoorva918) [reply pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/parquet/ParquetFileProcessor.java]: Done

File: scripts/run-with-gradle-raw-file.sh [hunk 2]
```diff
@@ -19,7 +19,7 @@ export KEYCLOAK_SERVICE_ACCOUNT_FILE=/opt/pravega-sensor-collector/conf/keycloak
 export JAVA_OPTS="-Xmx512m"
 
 export PRAVEGA_SENSOR_COLLECTOR_RAW1_CLASS=io.pravega.sensor.collector.rawfile.RawFileIngestService
-export PRAVEGA_SENSOR_COLLECTOR_RAW1_FILE_SPEC=/opt/pravega-sensor-collector/Files
+export PRAVEGA_SENSOR_COLLECTOR_RAW1_FILE_SPEC="/opt/pravega-sensor-collector/Files/A,/opt/pravega-sensor-collector/Files/B"
```
Comment (ShwethaSNayak) [inline scripts/run-with-gradle-raw-file.sh:22]: Along with Raw file are we adding multiple path for other file types like Parquet and CSV?

Comment (apoorva918) [reply scripts/run-with-gradle-raw-file.sh:22]: Yes, I've added this functionality in all 3 services

File: pravega-sensor-collector/src/test/resources/RawFileIngest.properties [hunk 3]
```diff
@@ -9,8 +9,8 @@
 #
 # This file can be used to manually test RawFileIngestService.
 PRAVEGA_SENSOR_COLLECTOR_RAW1_CLASS=io.pravega.sensor.collector.rawfile.RawFileIngestService
-PRAVEGA_SENSOR_COLLECTOR_RAW1_FILE_SPEC=/opt/pravega-sensor-collector/Files
-PRAVEGA_SENSOR_COLLECTOR_RAW1_FILE_EXTENSION=parquet
+PRAVEGA_SENSOR_COLLECTOR_RAW1_FILE_SPEC="/opt/pravega-sensor-collector/Files/A,/opt/pravega-sensor-collector/Files/B"
+PRAVEGA_SENSOR_COLLECTOR_RAW1_FILE_EXTENSION=txt
```
Comment (sachin-j-joshi) [inline pravega-sensor-collector/src/test/resources/RawFileIngest.properties:13]: should this be parquet? or empty?

**Review by sachin-j-joshi — APPROVED** [review]
LGTM.

One minor comment - Please add description to all public classes.

File: pravega-sensor-collector/src/test/resources/RawFileIngest.properties (hunk 3 above)
Comment (apoorva918) [reply pravega-sensor-collector/src/test/resources/RawFileIngest.properties]: Changed it to parquet

**Pull request merged**